cardano-node/test/Test/Cardano/Node/FilePermissions.hs

```diff
@@ -66,7 +67,7 @@ createFileWithOwnerPermissions :: HasTextEnvelope a => File () Out -> a -> Prope
 createFileWithOwnerPermissions targetfp value = do
   result <- liftIO $ writeLazyByteStringFileWithOwnerPermissions targetfp $ textEnvelopeToJSON Nothing value
   case result of
-    Left err -> failWith Nothing $ displayError err
+    Left err -> failWith Nothing $ displayError @(FileError ()) err
```

Comment:
The compiler needs to infer a concrete type for `e` because it "needs" to display the error.  The compile doesn't actually need the display the error because the constructor that has `e` is never used, but the compiler doesn't know that.  We have to help the compiler by fixing `e = ()` because there is an `Error ()` instance this works but it's unpleasant we have to help the compiler display an error that never gets displayed.